endpoints/openrtb2/auction.go

```diff
@@ -1389,22 +1392,38 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 					err = fmt.Errorf("imp.ext.prebid.storedrequest.id %s: Stored Imp has Invalid JSON: %s", impIds[i], Err)
 				}
 			}
-			return nil, []error{err}
+			return nil, nil, []error{err}
 		}
 		imps[idIndices[i]] = resolvedImp
+
+		impId, err := jsonparser.GetString(resolvedImp, "id")
```

Comment:
At this point, I don't think we've parsed the `resolvedImp` yet. Could the stored impression contain the imp id?

Reply:
`Could the stored impression contain the imp id? ` - this is possible. I take id from final impression data, this id can come from request.imp[] or stored impression.

Reply:
> At this point, I don't think we've parsed the resolvedImp yet

But. I don't think we need the resolvedImp to get the impID. In fact, we use the impID to get the stored Imp and construct the resolved Imp. See L1383. We have an `impsIds` array that we're looping on in this loop so why not just get the imp Id from that array as `impIds[i]`? Am I missing something here which blocks us from doing that?

Reply:
`impId` in L1383 is not imp id, it's stored imp id. It works together with `idIndices` which specifies what imp index has stored impId. This is extra variable to support requests where not all imps have stored imps. 
Please take a look at the attached screenshot. 
`for` loop in line 1383 iterates over all found stored imp ids. In my request example there are 3 impressions and first doesn't have stored imp id, but second ant third impressions have it. 
 - on iteration 0 it takes stored imp id `3c5416a2-e17a-4c9a-9379-3f74fd3fc378` and impression with index 1 from idIndexes. And merge them together. 
 - on iteration 1 it takes stored imp id `3c5416a2-e17a-4c9a-9379-3f74fd3fc378` and impression with index 2 from idIndexes. And merge them together. 
 
If we want to use `impIds[i]` we also need additional data to map it to impression id where it was used.
Using map here and in further processing is easier to understand than this `extra connection` variable. 
Hope this makes sense.
<img width="728" alt="Screen Shot 2021-08-02 at 4 52 07 PM" src="https://user-images.githubusercontent.com/4790765/127937861-428137fa-ff65-47f5-932d-d7c03807d66f.png">

Reply:
In my request imp ids were 1_1, 1_2 and 1_3. At the moment of merge we don't care about imp id because it can be overwritten by stored imp.  But we need to know the final impId to map it stored data.

Reply:
I see. Makes sense. Thanks for that clarification. That variable name is deceiving :D